lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -298,13 +301,41 @@ static LogicalResult applyConventionAnno(const AnnoPathValue &target,
 
   auto convention = *conventionOpt;
 
+  if (convention == Convention::Internal)
+    // Convention is internal by default so there is nothing to change
+    return success();
+
+  auto includeHierarchy = anno.getAs<BoolAttr>("includeHierarchy");
+  auto conventionAttr = ConventionAttr::get(op->getContext(), convention);
+  auto setConvention = [&](Operation *moduleOp) {
+    TypeSwitch<Operation *>(moduleOp)
+        .Case<FModuleOp, FExtModuleOp>([&](auto moduleOp) {
+          if (IsConventionAnno)
+            moduleOp.setConventionAttr(conventionAttr);
+          else
+            moduleOp->setDiscardableAttr("body_type_lowering", conventionAttr);
+        })
+        .Default([](auto) {});
+  };
```

Comment:
The `TypeSwitch` is kind of overkill here. Is there a trait that this can be cast to instead?

Unrelated, the checking earlier in this function does not guarantee that this is only applied to an `FModuleOp` or FExtModuleOp`. It only checks for `OpAnnoTarget` and then incorrectly says that is a module.

This also does bring up a point of if the internal convention should be allowed to be applied to external modules. It seems like _no_. However, that would mean that duplication as opposed to templating may be better.

Reply:
>  Is there a trait that this can be cast to instead?

I thought FModuleLike was meant to be the interface for that, but I'll just keep the current logic since FModuleLike is implemented for Class or MemModule and the behavior is unclear.  

> This also does bring up a point of if the internal convention should be allowed to be applied to external modules. It seems like no. However, that would mean that duplication as opposed to templating may be better.

Thanks, I changed to duplicate the handler.